### PR TITLE
fix build problem on macOS

### LINKED ISSUE #7: fail to build on macOS
There are several problems in the Makefile

```bash

$ rm -f parser.go; make 

make parser
bin/goyacc -o /dev/null parser.y
make[1]: bin/goyacc: No such file or directory
make[1]: *** [parser] Error 1
make: *** [parser.go] Error 2


make parser
bin/goyacc -o /dev/null parser.y
Parse table entries: 758041 of 1909026, x 16 bits == 1516082 bytes
bin/goyacc -o parser.go parser.y 2>&1 | egrep "(shift|reduce)/reduce" | awk '{print} END {if (NR > 0) {print "Find conflict in parser.y. Please check y.output for more information."; exit 1;}}'
rm -f y.output
sed: -e: No such file or directory
make[1]: *** [parser] Error 1
make: *** [parser.go] Error 2
```

### PR DESCRIPTION
**Pull request by darren**
this fixes #7 

1. there's a directory named `goyacc`, so make thinks goyacc already exists and will not build bin/goyacc

2. goyacc dependecies are not fetched

3.  ~~ARCH, LINUX MAC not defined~~

### CODE REVIEW

**Comment by darren** [comment]
PTAL @XuHuaiyu @tiancaiamao

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,8 @@ parser: goyacc
 
 	@awk 'BEGIN{print "// Code generated by goyacc"} {print $0}' parser.go > tmp_parser.go && mv tmp_parser.go parser.go;
 
-goyacc:
+goyacc-bin:
+	go get ./goyacc/
```
[the remark this answers is not in the document]
Comment (darren) [reply Makefile]: goyacc depends cznic's libraries and not vendored.

try this:
```bash 
export GOPATH=/tmp/ 
go get  github.com/pingcap/parser 
cd $GOPATH/src/github.com/pingcap/parser
rm -f parser.go
make
```

[the remark this answers is not in the document]
Comment (kennytm) [reply Makefile]: @darren The dependencies should be automatically fetched when `GO111MODULE=on` is set

[the remark this answers is not in the document]
Comment (darren) [reply Makefile]: @kennytm 
Got it!

**Pull request closed without merging**